molt/src/commands.rs

```diff
@@ -1189,6 +1189,76 @@ pub fn cmd_string_length(_interp: &mut Interp, _: ContextID, argv: &[Value]) ->
     molt_ok!(len)
 }
 
+/// string map ?-nocase? *charMap* *string*
+pub fn cmd_string_map(_interp: &mut Interp, _: ContextID, argv: &[Value]) -> MoltResult {
+    check_args(2, argv, 4, 5, "?-nocase? charMap string")?;
+
+    let mut nocase = false;
+
+    if argv.len() == 5 {
+        let opt = argv[2].as_str();
+
+        if opt == "-nocase" {
+            nocase = true;
+        } else {
+            return molt_err!("bad option \"{}\": must be -nocase", opt);
+        }
+    }
+
+    let char_map = argv[argv.len() - 2].as_dict()?;
+    let string = argv[argv.len() - 1].as_str();
+
+    let filtered_keys = char_map
+        .iter()
+        .map(|(k, v)| {
+            let new_k = if nocase {
+                Value::from(k.as_str().to_lowercase())
+            } else {
+                k.clone()
+            };
+
+            let count = new_k.as_str().chars().count();
+
+            (new_k, count, v.clone())
+        })
+        .filter(|(_, count, _)| *count > 0)
+        .collect::<Vec<_>>();
+    let matching_string = if nocase {
+        string.to_lowercase()
+    } else {
+        string.to_string()
+    };
+
+    let mut result = "".to_string();
```

Comment:
Probably better as `let mut result = String::new();`

Reply:
Done in 1c900a0.

Reply:
Done in 1c900a0.